plugins/modules/device_credential_workflow_manager.py

```diff
@@ -1551,8 +1555,7 @@ def get_want_device_credentials(self, CredentialDetails):
                         update_credential[update_cli_ptr] \
                             .update({"password": item.get("password")})
                     else:
-                        self.msg = "password is mandatory for udpating \
-                                   cliCredential " + str(have_cli_ptr)
+                        self.msg = "password is mandatory for udpating cliCredential " + str(have_cli_ptr)
```

Comment:
The password is mandatory for **updating** the cli_credential ... Can you check the other log messages and update?